codice/ddf
# DDF-3636 Fix logging regression where no error is logged when features fail to start

#### What does this PR do?
Adds error logging for the Jolokia servlet. 
This addresses lack of logging when feature resolution has an error. 

#### Who is reviewing it? 
@tbatie 
@GabrielFabian 

#### Choose 2 committers to review/merge the PR. 
@stustison 
@lessarderic

#### How should this be tested?
Install DDF. Ensure everything works as expected for a correct case. 

Remove the `!com.sun.management` statement from the pom of `catalog-ui-search` and ensure that the install process fails with an exception in the backend error logs. The stacktrace returned by Jolokia to the frontend should be redacted (use network tab in Chrome dev tools). 

#### Any background context you want to provide?
App decoupling has hid the Karaf features service from being exposed. Starting features is now done through Karaf's MBean directly, but it does not log the info we need, and this leaves no location in our code to catch exceptions that might occur. 

#### What are the relevant tickets?
[DDF-3636](https://codice.atlassian.net/browse/DDF-3636)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
